spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/RetryableConfigServicePropertySourceLocator.java

```diff
@@ -0,0 +1,72 @@
+package org.springframework.cloud.config.client;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.core.env.CompositePropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.retry.RetryCallback;
+import org.springframework.retry.RetryContext;
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.policy.NeverRetryPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+public class RetryableConfigServicePropertySourceLocator extends
+		AbstractConfigServicePropertyLocator {
+
+	private static Log logger = LogFactory
+			.getLog(RetryableConfigServicePropertySourceLocator.class);
+
+	private RetryTemplate retryTemplate;
+
+	public RetryableConfigServicePropertySourceLocator(ConfigClientProperties defaults) {
+		super(defaults);
+	}
+
+	@Override
+	protected PropertySource<?> tryLocating(final ConfigClientProperties client,
+			final CompositePropertySource composite, final Environment environment)
+			throws Exception {
+		ConfigClientProperties defaults = getDefaults();
+
+		if (defaults.getRetryBeforeFail() == 0) {
+			if (logger.isWarnEnabled()) {
+				logger.warn("To control retry count you can set the "
+						+ ConfigClientProperties.PREFIX + ".retryBeforeFail property");
+			}
+		}
+
+		return getRetryTemplate(defaults).execute(
+				new RetryCallback<PropertySource<?>, Exception>() {
+
+					@Override
+					public PropertySource<?> doWithRetry(RetryContext retryContext)
+							throws Exception {
+						return exchange(client, composite);
+					}
+				});
+	}
+
+	public void setRetryTemplate(RetryTemplate retryTemplate) {
+		this.retryTemplate = retryTemplate;
+	}
+
+	private RetryTemplate getRetryTemplate(ConfigClientProperties client) {
+		RetryTemplate retryTemplate = new RetryTemplate();
+		retryTemplate.setRetryPolicy(new NeverRetryPolicy());
+
+		if (client.getRetryBeforeFail() > 0) {
```

Comment:
I think the formatting I got from https://github.com/spring-cloud/spring-cloud-build/blob/master/spring-cloud-build/eclipse-code-formatter.xml is either not accurate or their is something wrong with the eclipse formatter plugin for IDEA.